impl/src/test/java/io/kroxylicious/testing/kafka/testcontainers/TestcontainersKafkaClusterTest.java

```diff
@@ -0,0 +1,256 @@
+                        assertThat(perImagePullPolicy.dockerImageName().getVersionPart()).isEqualTo("latest-kafka-3.7.0");
+                        final ImagePullPolicy actualPullPolicy = perImagePullPolicy.pullPolicy();
+                        assertThat(actualPullPolicy).isInstanceOf(ImagePullPolicy.class); // We can't see `AlwaysPullPolicy` as it's not public
+                        assertThat(actualPullPolicy.shouldPull(perImagePullPolicy.dockerImageName())).isTrue();
+                    });
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("fixedVersions")
+    void shouldAllowConfigToControlFixedKafkaVersion(Version version) {
+        KafkaClusterConfig config = clusterConfigBuilder.kafkaVersion(version.value()).build();
+        try (TestcontainersKafkaCluster testcontainersKafkaCluster = new TestcontainersKafkaCluster(config)) {
+
+            // When
+            final TestcontainersKafkaCluster.PerImagePullPolicy kafkaImage = testcontainersKafkaCluster.getKafkaImage();
+
+            // Then
+            assertThat(kafkaImage)
+                    .isNotNull().satisfies(perImagePullPolicy -> {
+                        assertThat(perImagePullPolicy.dockerImageName().getVersionPart()).isEqualTo("latest-kafka-" + version.value());
+                        final ImagePullPolicy actualPullPolicy = perImagePullPolicy.pullPolicy();
+                        assertThat(actualPullPolicy).isInstanceOf(ImagePullPolicy.class); // We can't see `AlwaysPullPolicy` as it's not public
+                        assertThat(actualPullPolicy.shouldPull(perImagePullPolicy.dockerImageName())).isTrue();
+                    });
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("floatingVersions")
+    void shouldAllowConfigToControlFloatingKafkaVersion(Version version) {
+        KafkaClusterConfig config = clusterConfigBuilder.kafkaVersion(version.value()).build();
+        try (TestcontainersKafkaCluster testcontainersKafkaCluster = new TestcontainersKafkaCluster(config)) {
+
+            // When
+            final TestcontainersKafkaCluster.PerImagePullPolicy kafkaImage = testcontainersKafkaCluster.getKafkaImage();
+
+            // Then
+            assertThat(kafkaImage)
+                    .isNotNull().satisfies(perImagePullPolicy -> {
+                        assertThat(perImagePullPolicy.dockerImageName().getVersionPart()).isEqualTo(version.value());
+                        final ImagePullPolicy actualPullPolicy = perImagePullPolicy.pullPolicy();
+                        assertThat(actualPullPolicy).isInstanceOf(ImagePullPolicy.class); // We can't see `AlwaysPullPolicy` as it's not public
+                        assertThat(actualPullPolicy.shouldPull(perImagePullPolicy.dockerImageName())).isTrue();
+                    });
+        }
+    }
+
+    private static Stream<Version> fixedVersions() {
+        return Stream.of(
+                version("3.6.0"),
```

Comment:
Is there a reason to stop at 3.6.0?

Reply:
It's the same set as the TemplateTest which tries to actually fire things up (this test was written as that was failing and hard to debug).

I'm not sure the set of versions actually matters but while we are talking about it I'll add the missing versions.